eng/Publishing.props

```diff
@@ -0,0 +1,31 @@
+<Project>
+
+  <PropertyGroup>
+    <PublishDependsOnTargets>$(PublishDependsOnTargets);CollectPackageChecksumFiles</PublishDependsOnTargets>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageChecksumFile Include="$(ArtifactsShippingPackagesDir)**\*.sha512" IsShipping="true" />
+    <PackageChecksumFile Include="$(ArtifactsNonShippingPackagesDir)**\*.sha512" IsShipping="false" />
+  </ItemGroup>
+
+  <Target Name="CollectPackageChecksumFiles">
+    <!-- Calculate the blob sub path for each checksum file. Checksum files do not necessarily have
+         an associated blog group, thus the sub path is optional. -->
+    <ItemGroup>
+      <!-- Read in blog group name -->
+      <PackageChecksumFile Update="@(PackageChecksumFile)">
+        <BlobSubPath>$([System.IO.File]::ReadAllText('%(RootDir)%(Directory)%(Filename).blobgroup'))/</BlobSubPath>
```

Comment:
The WriteLinesToFile task writes out a newline at the end of the content. The File::ReadAllText here reads in the content along with the newline, so the newline ends up in the BlobSubPath. I'll refactor this target to use ReadLinesFromFile task to read in the lines and use target batching on the PackageChecksumFile item group.